### PR TITLE
DBZ-504 Type changes detected for constraints

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-504

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java [hunk 1]
```diff
@@ -440,6 +440,22 @@ private boolean schemaChanged(List<ReplicationMessage.Column> columns, Table tab
                                 incomingType);
                     return true;
                 }
+                if (metadataInMessage) {
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java]: To be sure, we get length and scale from the (latest) wal2json and decoderbufs, correct?

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java [hunk 2]
```diff
@@ -394,7 +394,7 @@ private void assertValue(Struct content) {
 
         private void assertStruct(final Struct expectedStruct, final Struct actualStruct) {
             expectedStruct.schema().fields().stream().forEach(field -> {
-                final Object expectedValue = actualStruct.get(field);
+                final Object expectedValue = expectedStruct.get(field);
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java]: Ouch, good spot!

**Comment by gunnarmorling** [comment]
@jpechane One thing I'm curious about is that it doesn't seem to require any changes for DecoderBufs. So that info was already contained in the messages as per your recent changes, and it also is read at the connector side?

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java]: Yes, from the latest - unfrotunately we can remove it only after RDS will move forward with `wal2json` version

**Comment by jpechane** [comment]
Yes, all the data were added in time of 0.7. This was a follow-up task to cover what was not completed yet neither in 0.7 nor in versions before.

**Comment by gunnarmorling** [comment]
LGTM, rebased and applied. Thanks, @jpechane!

**Pull request merged**